controllers/machine_controller_noderef.go

```diff
@@ -89,16 +88,12 @@ func (r *MachineReconciler) reconcileNodeRef(_ context.Context, cluster *cluster
 	return nil
 }
 
-func (r *MachineReconciler) getNodeReference(client client.Client, providerID *noderefutil.ProviderID) (*apicorev1.ObjectReference, error) {
+func (r *MachineReconciler) getNodeReference(c client.Client, providerID *noderefutil.ProviderID) (*apicorev1.ObjectReference, error) {
 	logger := r.Log.WithValues("providerID", providerID)
 
-	listOpt := metav1.ListOptions{}
-
+	nodeList := apicorev1.NodeList{}
 	for {
-		nodeList := apicorev1.NodeList{}
-		// TODO Add a context to this method
-		err := client.List(context.TODO(), &nodeList)
-		if err != nil {
+		if err := c.List(context.TODO(), &nodeList, client.Continue(nodeList.Continue)); err != nil {
```

Comment:
I guess this works if the continue token is the empty string (i.e. it performs the initial list)?

Reply:
Yeah, I looked in the default value of the option field set by `client.Continue`, and it's an empty string